pkg/ctl/cmdutils/resource.go

```diff
@@ -26,14 +27,49 @@ type ResourceCmd struct {
 	Include, Exclude []string
 }
 
+// NewCtl performs common defaulting and validation and constructs a new
+// instance of eks.ClusterProvider, it may return an error if configuration
+// is invalid or region is not supported
+func (rc *ResourceCmd) NewCtl() (*eks.ClusterProvider, error) {
+	api.SetClusterConfigDefaults(rc.ClusterConfig)
+
+	if err := api.ValidateClusterConfig(rc.ClusterConfig); err != nil {
+		if rc.Validate {
+			return nil, err
+		}
+		logger.Warning("ignoring validation error: %s", err.Error())
+	}
+
+	for i, ng := range rc.ClusterConfig.NodeGroups {
+		if err := api.ValidateNodeGroup(i, ng); err != nil {
+			if rc.Validate {
+				return nil, err
+			}
+			logger.Warning("ignoring validation error: %s", err.Error())
+		}
+		// defaulting of nodegroup currently depends on validation;
+		// that may change, but at present that's how it's meant to work
+		api.SetNodeGroupDefaults(i, ng)
+	}
+
+	ctl := eks.New(rc.ProviderConfig, rc.ClusterConfig)
```

Comment:
This could be something like:
```
if rc.MockProvider != nil {
  return rc.MockProvider, nil
}

ctl := eks.New(rc.ProviderConfig, rc.ClusterConfig)
```

Reply:
But let's leave until next iteration, I also have a PR to rename `rc` to simply `cmd`, which I promised to Martina a long time ago.